tcms/rpc/tests/test_testplan.py

```diff
@@ -226,3 +227,116 @@ def test_ignores_existing_mappings(self):
+
+    def test_create_plan_without_permissions(self):
+        unauthorized_user = UserFactory()
+        unauthorized_user.set_password('api-testing')
+        unauthorized_user.save()
+
+        unauthorized_user.user_permissions.add(*Permission.objects.all())
+        remove_perm_from_user(unauthorized_user, 'testplans.add_testplan')
+
+        rpc_client = xmlrpc.TCMSXmlrpc(unauthorized_user.username,
+                                       'api-testing',
+                                       '%s/xml-rpc/' % self.live_server_url).server
+
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            rpc_client.TestPlan.create(self.params)
+
+
+class TestListAttachments(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan = TestPlanFactory(product_version=self.version,
+                                    product=self.product,
+                                    author=UserFactory(),
+                                    type=PlanTypeFactory(name='manual smoking'))
+
+    def test_list_attachments(self):
+        attachments = self.rpc_client.TestPlan.list_attachments(self.plan.pk)
+        self.assertIsNotNone(attachments)
+        self.assertIsInstance(attachments, list)
+
+    def test_list_attachments_with_wrong_plan_id(self):
+        with self.assertRaisesRegex(XmlRPCFault, 'TestPlan matching query does not exist'):
+            self.rpc_client.TestPlan.list_attachments(-1)
+
+    def test_list_attachments_without_permissions(self):
+        unauthorized_user = UserFactory()
+        unauthorized_user.set_password('api-testing')
+        unauthorized_user.save()
+
+        unauthorized_user.user_permissions.add(*Permission.objects.all())
+        remove_perm_from_user(unauthorized_user, 'attachments.view_attachment')
+
+        rpc_client = xmlrpc.TCMSXmlrpc(unauthorized_user.username,
+                                       'api-testing',
+                                       '%s/xml-rpc/' % self.live_server_url).server
+
+        with self.assertRaisesRegexp(ProtocolError, '403 Forbidden'):
+            rpc_client.TestPlan.list_attachments(self.plan.pk)
+
+
+class TestAddAttachment(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan = TestPlanFactory(product_version=self.version,
+                                    product=self.product,
+                                    author=UserFactory(),
+                                    type=PlanTypeFactory(name='manual smoking'))
+
+    def test_add_attachment(self):
+        self.rpc_client.TestPlan.add_attachment(self.plan.pk, 'attachment.txt', 'kiwitcms')
+
+    def test_add_attachment_without_permissions(self):
```

Comment:
This is not a good way to test around permissions. Use `APIPermissionsTestCase`, see #1716.

Reply:
I actually used same implementation in the existing cases, for instance https://github.com/kiwitcms/Kiwi/blob/174f4ebd579222f92f0a0c778436d729680c4fad/tcms/rpc/tests/test_testplan.py#L67
If there is no concern I would update them as well.

Reply:
I can see that, but this implementation is old and not very accurate. As of yesterday there is a newer one which every future test should follow.

> If there is no concern I would update them as well.

Feel free to do so but do it as separate pull request(s) so they can be reviewed and merged independently.